src/quantile.rs

```diff
@@ -369,6 +423,28 @@ where
         Some(current_pattern_min)
     }
 
+    fn argmin_skipnan(&self) -> Option<D::Pattern>
+    where
+        A: MaybeNan,
+        A::NotNan: Ord,
+    {
+        let mut pattern_min = D::zeros(self.ndim()).into_pattern();
+        let min = self.indexed_fold_skipnan(None, |current_min, (pattern, elem)| {
+            Some(match current_min {
+                Some(m) if m <= elem => m,
```

Comment:
`if m <= elem => m` take me a while to get this 😂

Reply:
Lol, I didn't notice that. Maybe parentheses would help? `Some(m) if (m <= elem) => m,`

rust-lang/rfcs#2497 would be nice here so that we could replace the `Some(match ...` with `if let`:

```rust
            if let Some(m) = current_min && m <= elem {
                Some(m)
            } else {
                pattern_min = pattern;
                Some(elem)
            }
```

Reply:
Yeah either way (perhaps better with `if let`) would be clearer than the current code!

Reply:
Yeah, I'm looking forward to RFC 2497 being implemented. :) For the time being, let's add the parentheses.